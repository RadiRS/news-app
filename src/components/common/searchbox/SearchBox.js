import React from 'react';
import { KeyboardAvoidingView } from 'react-native';
import styled from 'styled-components/native';
import Icon from '../icon';
import { Colors } from '../../../themes';

const SearchBox = props => {
  const { onPress, onChangeText, placeholder, disabled } = props;
  return (
    <KeyboardAvoidingView>
      <Container>
        <TextInput onChangeText={onChangeText} placeholder={placeholder} />
        <Button onPress={onPress} disabled={disabled}>
          <Icon name="ios-search" />
        </Button>
      </Container>
    </KeyboardAvoidingView>
  );
};

export default SearchBox;

const Container = styled.View`
  elevation: 2px;
  flex-direction: row;
  height: 60px;
  border-width: 2px;
  border-color: ${Colors.cloud};
  border-radius: 14px;
  margin: 10px 20px;
  padding: 5px;
  align-items: center;
  /* justify-content: center; */
`;

const TextInput = styled.TextInput`
  font-size: 18px;
  background-color: white;
  flex: 0.9;
  border-radius: 14px;
`;

const Button = styled.TouchableOpacity`
  flex: 0.1;
  align-items: center;
  justify-content: center;
`;
