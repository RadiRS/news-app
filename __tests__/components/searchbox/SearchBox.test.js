import 'react-native';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import SearchBox from '../../../src/components/common/searchbox';

test('renders SearchBox component correctly', () => {
  const tree = renderer
    .create(
      <SearchBox onPress={() => ''} onChangeText={() => ''} placeholder="" />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
