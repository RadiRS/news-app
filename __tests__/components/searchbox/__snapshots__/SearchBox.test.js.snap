// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SearchBox component correctly 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "rgba(200,200,200, 0.35)",
          "borderRadius": 14,
          "borderWidth": 2,
          "elevation": 2,
          "flexDirection": "row",
          "height": 60,
          "marginBottom": 10,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 10,
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 5,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder=""
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderRadius": 14,
            "flexBasis": 0,
            "flexGrow": 0.9,
            "flexShrink": 1,
            "fontSize": 18,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 0.1,
          "flexShrink": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#546bfb",
              "fontSize": 24,
            },
            Array [
              undefined,
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï’¥
      </Text>
    </View>
  </View>
</View>
`;
